staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go

```diff
@@ -156,13 +156,20 @@ func (ss *scaleSet) newVMSSVirtualMachinesCache() (*timedCache, error) {
 					}
 
 					computerName := strings.ToLower(*vm.OsProfile.ComputerName)
-					localCache.Store(computerName, &vmssVirtualMachinesEntry{
+					vmssVMCacheEntry := &vmssVirtualMachinesEntry{
 						resourceGroup:  resourceGroup,
 						vmssName:       ssName,
 						instanceID:     to.String(vm.InstanceID),
 						virtualMachine: &vm,
 						lastUpdate:     time.Now().UTC(),
-					})
+					}
+					// set cache entry to nil when the VM is under deleting.
+					if vm.VirtualMachineScaleSetVMProperties != nil &&
+						strings.EqualFold(to.String(vm.VirtualMachineScaleSetVMProperties.ProvisioningState), string(compute.ProvisioningStateDeleting)) {
```

Comment:
What happens if the vm that's in deleting state goes to failed state? Is that a possibility?

If that's possible wouldn't we want to keep the node as `NotReady` in the cluster, so the user is aware of the issue with the node?

Reply:
the deleting VM won't go to running again in any cases, hence it's safe to remove it from the cluster.

Reply:
ok, thanks!